File: python version/fine.py
from typing import List
from violation import Violation

class Fine:
    def __init__(self, plate_number: str, violations: List[Violation]):
        self.plate_number = plate_number
        self.violations = list(violations)

    def get_total_amount(self) -> float:
        return sum(v.fee for v in self.violations)

    def print_fine(self) -> None:
        print(f"Traffic for car {self.plate_number}")
        print(f"Total amount: {int(self.get_total_amount())} EGP")
        print("Violations:")
        for v in self.violations:
            print(f"- {v.description} : {int(v.fee)} EGP")

File: Fine.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Fine {
    private final String plateNumber;
    private final List<Violation> violations;

    public Fine(String plateNumber, List<Violation> violations) {
        this.plateNumber = plateNumber;
        this.violations = new ArrayList<>(violations);
    }

    public String getPlateNumber() {
        return plateNumber;
    }

    public List<Violation> getViolations() {
        return Collections.unmodifiableList(violations);
    }

    public double getTotalAmount() {
        double total = 0;
        for (Violation v : violations) {
            total += v.getFee();
        }
        return total;
    }

    public void printFine() {
        System.out.println("Traffic for car " + plateNumber);
        System.out.println("Total amount: " + (int) getTotalAmount() + " EGP");
        System.out.println("Violations:");
        for (Violation v : violations) {
            System.out.println("- " + v.getDescription() + " : " + (int) v.getFee() + " EGP");
        }
    }
}
